pkg/parcacol/querier.go

```diff
@@ -1457,6 +1457,110 @@ func isSamplesCount(st profile.ValueType) bool {
 	return st.Type == "samples" && st.Unit == "count"
 }
 
+func (q *Querier) GetProfileMetadataMappings(
+	ctx context.Context,
+	query string, startTime, endTime time.Time,
+) ([]string, error) {
+	ctx, span := q.tracer.Start(ctx, "Querier/MappingFiles")
+	defer span.End()
+
+	_, selectorExprs, err := QueryToFilterExprs(query)
+	if err != nil {
+		return nil, err
+	}
+
+	start := timestamp.FromTime(startTime)
+	end := timestamp.FromTime(endTime)
+	filterExpr := logicalplan.And(
+		append(
+			selectorExprs,
+			logicalplan.Col(profile.ColumnTimestamp).GtEq(logicalplan.Literal(start)),
+			logicalplan.Col(profile.ColumnTimestamp).LtEq(logicalplan.Literal(end)),
+		)...,
+	)
+
+	records := make(map[string]struct{})
+	err = q.engine.ScanTable(q.tableName).
+		Filter(filterExpr).
+		Project(logicalplan.Col("stacktrace")).
+		Execute(ctx, func(ctx context.Context, r arrow.Record) error {
+			r.Retain()
+
+			locations := r.Column(0).(*array.List)
+
+			values := locations.ListValues().(*array.Dictionary)
+			valueDict := values.Dictionary().(*array.Binary)
+
+			for i := 0; i < values.Len(); i++ {
+				encodedLocation := valueDict.Value(values.GetValueIndex(i))
+				symInfo, _ := profile.DecodeSymbolizationInfo(encodedLocation)
+				records[symInfo.Mapping.File] = struct{}{}
+			}
+
+			return nil
+		})
+	if err != nil {
+		return nil, err
+	}
+
+	res := make([]string, 0, len(records))
+	for r := range records {
+		res = append(res, r)
+	}
+
+	return res, nil
```

Comment:
I quite like the idea of sorting the labels, let's do the same with the binaries?